WcaOnRails/app/views/persons/_results_by_event.html.erb

```diff
@@ -0,0 +1,61 @@
+<div class="results-by-event">
+  <div class="event-selector text-center">
+    <% @results.map(&:event).uniq.each do |event| %>
+      <span class="event-radio">
+        <%= label_tag "radio-#{event.id}" do %>
+          <%= radio_button_tag "results_event", event.id, params[:results_event] == event.id, id: "radio-#{event.id}" %>
+          <%= cubing_icon event.id, data: { toggle: "tooltip", placement: "top" }, title: event.name %>
+        <% end %>
+      </span>
+    <% end %>
+  </div>
+  <div class="table-responsive">
+    <table class="table table-striped floatThead">
+      <thead>
+        <tr>
+          <th class="competition"><%= t 'persons.show.competition' %></th>
+          <th class="round"><%= t 'competitions.results_table.round' %></th>
+          <th class="place"><%= t 'persons.show.place' %></th>
+          <th class="single"><%= t 'common.single' %></th>
+          <th class="regional-single-record"></th>
+          <th class="average"><%= t 'common.average' %></th>
+          <th class="regional-average-record"></th>
+          <th class="solves" colspan="5"><%= t 'common.solves' %></th>
+        </tr>
+      </thead>
+      <% @results.group_by(&:event).each do |event, results| %>
+        <tbody class="event-<%= event.id %>">
+          <tr>
+            <td colspan="12" class="event">
+              <%= cubing_icon event.id %>
+              <%= t "events.#{event.id}" %>
+            </td>
+          </tr>
+          <% pb_markers = pb_markers(results.reverse) # Compute pb markers for results in the chronological order. %>
+          <% results.group_by(&:competitionId).each do |_, results| %>
```

Comment:
Similar question as before... is there any guarantee about the order of the keys in the hash returned by `group_by`?